File: src/main/java/gui/component/inputTabs/RandomParametersTab.java
package gui.component.inputTabs;

import data.Line;
import data.model.TrendLine;
import javafx.fxml.FXML;
import javafx.scene.control.TextField;
import javafx.scene.control.TextFormatter;

import java.text.NumberFormat;

public class RandomParametersTab extends ParametersTab {
    @FXML
    private TextField dotsField;
    @FXML
    private TextField seedField;

    private boolean dotsInvalid = true, seedInvalid = true;

    public RandomParametersTab() {
        super();
        this.setText("Cлучайный");
        NumberFormat nf = NumberFormat.getNumberInstance();
        nf.setGroupingUsed(false);
        loadFxml(getClass().getResource("RandomParametersTab.fxml"),this);
        dotsField.setTextFormatter(new TextFormatter<>(change ->
                (change.getControlNewText().matches("([1-9][0-9]*)?")) ? change : null)); //accept only unsigned integer
        dotsField.textProperty().addListener((observable, oldValue, newValue) -> {
            dotsInvalid = newValue.length() == 0 || newValue.length() > 6;
            if(!dotsInvalid) {
                dotsField.getStyleClass().remove("error");
            } else {
                dotsField.getStyleClass().add("error");
            }
            setIsInvalid(dotsInvalid || seedInvalid);
        });
        seedField.setTextFormatter(new TextFormatter<>(change ->
                (change.getControlNewText().matches("([1-9][0-9]*)?")) ? change : null)); //accept only unsigned integer
        seedField.textProperty().addListener((observable, oldValue, newValue) -> {
            seedInvalid = newValue.length() == 0 || newValue.length() > 6;
            if(!seedInvalid) {
                seedField.getStyleClass().remove("error");
            } else {
                seedField.getStyleClass().add("error");
            }
            setIsInvalid(seedInvalid || dotsInvalid);
        });
    }

    @Override
    public Line generateResult() {
        return TrendLine.Random(Integer.parseUnsignedInt(dotsField.getText()),
                Integer.parseInt(seedField.getText()));
    }
}

File: src/main/java/gui/component/inputTabs/RandomParametersTab.kt
package gui.component.inputTabs

import core.input.LineGenerator
import gui.component.control.ParametersTab
import gui.component.or
import gui.component.control.TextFieldFormat
import gui.component.control.ValidatableTextField

class RandomParametersTab : ParametersTab("Cлучайный") {
    private val dots = ValidatableTextField("Всего точек", TextFieldFormat.UINT)
    private val seed = ValidatableTextField("Значение инициализации", TextFieldFormat.UINT)

    init {
        getExtension().addAll(dots, seed)
        isInvalidField.bind(dots.isInvalidProp or seed.isInvalidProp)
    }

    override fun generateResult() = listOf(
            LineGenerator.random(dots.text.toInt(), 0.0, 1.0, seed.text.toInt())
    )
}
